libs/common/src/sql-db-entities/utilisation-report/utilisation-report.entity.ts

```diff
@@ -79,4 +79,14 @@ export class UtilisationReportEntity extends AuditableBaseEntity {
     this.data = data;
     this.updatedByUserId = getDbAuditUpdatedByUserId(requestSource);
   }
+
+  public setAsCompleted({ requestSource }: { requestSource: DbRequestSource }): void {
+    this.status = 'RECONCILIATION_COMPLETED';
+    this.updatedByUserId = getDbAuditUpdatedByUserId(requestSource);
+  }
+
+  public setAsNotCompleted({ requestSource }: { requestSource: DbRequestSource }): void {
+    this.status = this.azureFileInfo ? 'PENDING_RECONCILIATION' : 'REPORT_NOT_RECEIVED';
```

Comment:
I think it would be better for this choice of status to be handled explicitly in the state machine, and keep the methods on the entity class to `setAsState` or `setStatus` which accepts a status parameter. This feels slightly hidden to me and I think it would make sense for the state machine event handler that gets called to handle the business logic around states, especially as we build out logic around further state transitions with more complex interactions.